src/core/core_i_framework.nss

```diff
@@ -1144,6 +1149,15 @@ int RunEvent(string sEvent, object oInit = OBJECT_INVALID, object oSelf = OBJECT
             break;
     }
 
+    // Run tag-based scripts for any object, items are already handled
+    if (oSelf != GetModule())
+    {
+        int nObjectType = GetObjectType(oSelf);
+        if (ENABLE_TAGBASED_SCRIPTS && !(nState & EVENT_STATE_ABORT) &&
+            nObjectType < OBJECT_TYPE_INVALID)
```

Comment:
Is there a reason this isn't just using `GetIsObjectValid()`?

Reply:
That line was originally included to check if it was an item (!= OBJECT_TYPE_ITEM) since items are already handled separately. Looks like another result of inattention in review. Thanks for the catch, I'll fix this up tonight and resubmit.